pkg/controller/sshd/sshd_controller.go

```diff
@@ -0,0 +1,564 @@
+package sshd
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/go-logr/logr"
+
+	cloudingressv1alpha1 "github.com/openshift/cloud-ingress-operator/pkg/apis/cloudingress/v1alpha1"
+	"github.com/openshift/cloud-ingress-operator/pkg/awsclient"
+	"github.com/openshift/cloud-ingress-operator/pkg/config"
+
+	utils "github.com/openshift/cloud-ingress-operator/pkg/controller/utils"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_sshd")
+
+// Add creates a new SSHD Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileSSHD{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("sshd-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource SSHD
+	err = c.Watch(&source.Kind{Type: &cloudingressv1alpha1.SSHD{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Deployments
+	err = c.Watch(&source.Kind{Type: &appsv1.Deployment{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &cloudingressv1alpha1.SSHD{},
+	})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to Services
+	err = c.Watch(&source.Kind{Type: &corev1.Service{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &cloudingressv1alpha1.SSHD{},
+	})
+	if err != nil {
+		return err
+	}
```

Comment:
One caution is that the service and deployment (or ds, or whatever) need to be in the same namespace as the operator for these watches to work the way you want. Cross-namespace watches and controller references are not supported in Kubernetes.

Reply:
Right.  I think I've ensured that in the code.

Reply:
Talked this over with @cblecker and decided we'll just have to do without the watches.
The reconcile loop will still run periodically, and that's good enough for this controller.

Reply:
I want to revise this:

It seems like it should work as is so long as the CR is in the same namespace as the objects being watched. In this case, the CR will be in `openshift-sre-sshd` and the child resources will also be in the same namespace. If the CR was in the same namespace where the operator runs (`openshift-cloud-ingress-operator`), I think you'd hit these problems.

Reply:
Yes, that's the plan.  The CR and its child resources will be in `openshift-sre-sshd`.

Reply:
I'd add a note here that it's strange for an operator to create objects in distant namespaces, but that's literally what this entire operator is doing before this PR even came to be. ;-) But hey, it's cluster-scoped, so let's roll with it.

Reply:
Note added.